oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java

```diff
@@ -127,6 +128,13 @@ protected SplitDocumentCleanUp createCleanUp(Set<SplitDocType> gcTypes,
         // With OAK-8351 this switched from 1 to 2 queries (see createQueries)
         // hence we iterate over the queries returned by createQueries
         List<Bson> queries = createQueries(gcTypes, sweepRevs, oldestRevTimeStamp);
+        BasicDBObject keys = null;
+        if(hasIndex(getNodeCollection(), SD_TYPE, SD_MAX_REV_TIME_IN_SECS)) {
+           keys = new BasicDBObject();
+           keys.put(SD_TYPE,1);
+           keys.put(SD_MAX_REV_TIME_IN_SECS, 1);
+        }
+
```

Comment:
What do you think about moving this up to the constructor? That way, Revision GC will only check for the index once and now with every GC cycle.

Reply:
Marking this as resolved as the hint is now created in the constructor